package/tests/test_data_access/test_repositories/test_data_catalog.py

```diff
@@ -17,3 +17,20 @@ def test_get_layer_mapping_for_transcoded_dataset(self):
         repo.set_catalog(catalog)
         assert repo.get_layer_for_dataset("cars") == "raw"
         assert repo.get_layer_for_dataset("cars@pandas") == "raw"
+
+    def test_get_layer_mapping_from_metadata(self):
```

Comment:
I'd suggest adding another test to check that indeed a mix of `layer` definitions is supported.